### PR TITLE
#9788: Remove TopK from TTLib and replace all references with the TTNN api

### PR DESCRIPTION
**Pull request by sjameelTT**
### Ticket
#9788 

### Problem description
TTLib implementation needs to be replaced and moved across the codebase.

### What's changed
1) Remove the ttlib python api usages
2) Remove the ttlib tests
3) Remove the ttlib implementation

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by sjameelTT** [comment]
Post commit passes: https://github.com/tenstorrent/tt-metal/actions/runs/9763747265

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_topk.py
Comment (TT-BrianLiu) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_topk.py:1]: We have equivalent test in ttnn?

**Review by TT-BrianLiu — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by ayerofieiev-tt** [comment]
@sjameelTT , don't forget to update PR title :)

**Comment by sjameelTT** [comment]
> @sjameelTT , don't forget to update PR title :)

Lol done

**Comment by ayerofieiev-tt** [comment]
Screenshot 2024-07-02 at 12 24 04 PM

:P

**Review by mtairum — APPROVED** [review]
Tested branch locally. All good on the Mistral/Mixtral front.

Just make sure to run the t3k demo pipeline after rebasing to guarantee there are no regressions on latest main.

**Comment by sjameelTT** [comment]
Falcon7b is also failing on main and doesn't call this op, so the demo test failure is unrelated. Mixtral passes on the demo test.

**Pull request merged**